wodles/gcloud/tests/test_integration.py

```diff
@@ -8,82 +8,85 @@
+    """Test format_msg returns a well-formatted message."""
+    integration = WazuhGCloudIntegration(logger=MagicMock())
+    msg = integration.format_msg(json.dumps(test_message))
+    assert isinstance(msg, str)
+    msg_json = json.loads(msg)
+    assert 'integration' in msg_json
+    assert 'gcp' in msg_json
+    assert msg_json['gcp'] == test_message
 
 
 @patch('integration.socket.socket')
-def test_send_message_ok(mock_socket):
-    """Test if messages are sent to Wazuh queue socket."""
-    client = get_wazuhgcloud_subscriber()
-    with client.initialize_socket():
-        client.send_msg(test_message)
-    mock_socket.return_value.send.assert_called()
+def test_WazuhGCloudIntegration_initialize_socket(mock_socket):
+    """Test initialize_socket establish a connection with the ANALYSISD socket."""
+    integration = WazuhGCloudIntegration(logger=MagicMock())
+    integration.initialize_socket()
+    integration.socket.connect.assert_called_with(ANALYSISD)
 
 
-def test_send_message_ko():
-    """Test send_message method when the socket hasn't been initialized."""
-    with pytest.raises(AttributeError):
-        client = get_wazuhgcloud_subscriber()
-        client.send_msg(test_message)
+@pytest.mark.parametrize('raised_exception, expected_exception, errcode', [
+    (ConnectionRefusedError, exceptions.WazuhIntegrationInternalError, 1),
+    (OSError, exceptions.WazuhIntegrationInternalError, 2)
+])
+def test_WazuhGCloudIntegration_initialize_socket_ko(raised_exception, expected_exception, errcode):
+    """Test initialize_socket properly handles exceptions."""
+    integration = WazuhGCloudIntegration(logger=MagicMock())
+    with patch('socket.socket', side_effect=raised_exception), pytest.raises(expected_exception) as e:
+        integration.initialize_socket()
+    assert errcode == e.value.errcode
 
 
-def test_format_msg():
-    """Test if messages are formatted properly before to be sent."""
-    client = get_wazuhgcloud_subscriber()
-    formatted_message = client.format_msg(test_message)
+def test_WazuhGCloudIntegration_process_data():
+    """Test process_data is not implemented for this base class."""
+    integration = WazuhGCloudIntegration(logger=MagicMock())
+    with pytest.raises(NotImplementedError):
+        integration.process_data()
 
-    assert type(formatted_message) is str
+
+@patch('integration.socket.socket')
+def test_WazuhGCloudIntegration_send_message(mock_socket):
+    """Test if messages are sent to Wazuh queue socket."""
+    integration = WazuhGCloudIntegration(logger=MagicMock())
+    with integration.initialize_socket():
+        integration.send_msg(test_message)
+    mock_socket.return_value.send.assert_called()
 
 
 @pytest.mark.parametrize('raised_exception, expected_exception, errcode', [
-    (ConnectionRefusedError, exceptions.WazuhIntegrationInternalError, 1),
-    (OSError, exceptions.WazuhIntegrationInternalError, 2)
+    (OSError, exceptions.WazuhIntegrationInternalError, 3)
```

Comment:
No need to parametrize

Edit: if you leave the parameters for future new values, also leave docstring